src/sortStudents.ts

```diff
@@ -1,16 +1,77 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const studentsArray = JSON.parse(JSON.stringify(students));
+  const rightOrder = order === 'asc' ? 1 : -1;
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  switch (sortBy) {
+    case SortType.Name:
+      return studentsArray.sort((
+        prevStud: Student,
+        nextStud: Student,
+      ) => (rightOrder * prevStud.name.localeCompare(nextStud.name)));
+
+    case SortType.Surname:
+      return studentsArray.sort((
+        prevStud: Student,
+        nextStud: Student,
+      ) => (rightOrder * prevStud.surname.localeCompare(nextStud.surname)));
```

Comment:
```suggestion
    case SortType.Name:
    case SortType.Surname:
      return studentsArray.sort((
        prevStud: Student,
        nextStud: Student,
      ) => (rightOrder * prevStud[sortBy].localeCompare(nextStud[sortBy])));
```
checklist p.3 [CODE STYLE] - DRY. Combine logic for sorting by fields with the same data types.

Do the same for Age and Married